ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java

```diff
@@ -338,4 +370,56 @@ private void addStatTask(ASTNode root, Table table, Path oldPartitionLocation, P
       moveTask.addDependentTask(statTask);
     }
   }
+
+  public StringBuilder constructDeleteQuery(Table table, Map<String, String> partitionSpec) throws SemanticException {
+    String qualifiedTableName = HiveTableName.ofNullable(HiveUtils.unparseIdentifier(table.getTableName(), this.conf),
+            HiveUtils.unparseIdentifier(table.getDbName(), this.conf), null).getNotEmptyDbTable();
+    StringBuilder sb = new StringBuilder().append("delete from ").append(qualifiedTableName)
+            .append(" where ");
+    Deserializer deserializer = table.getDeserializer();
+    Map<String, PrimitiveObjectInspector.PrimitiveCategory> stringTypeInfoMap = new HashMap<>();
+    try {
+      ObjectInspector objectInspector = deserializer.getObjectInspector();
+      if (objectInspector.getCategory() == ObjectInspector.Category.STRUCT) {
+        StructObjectInspector structObjectInspector = (StructObjectInspector) objectInspector;
+        List<? extends StructField> structFields =  structObjectInspector.getAllStructFieldRefs();
+        for (StructField structField : structFields) {
+          if (structField.getFieldObjectInspector().getCategory() == ObjectInspector.Category.PRIMITIVE) {
+            PrimitiveObjectInspector primitiveObjectInspector = (PrimitiveObjectInspector) structField.getFieldObjectInspector();
+            stringTypeInfoMap.put(structField.getFieldName(),
+                primitiveObjectInspector.getTypeInfo().getPrimitiveCategory());
+          }
+        }
+      }
+    } catch (SerDeException e) {
+      throw new SemanticException(String.format("Unable to get object inspector due to: %s", e));
+    }
+    boolean first = true;
+    for (String key : partitionSpec.keySet()) {
+      PrimitiveObjectInspector.PrimitiveCategory category = stringTypeInfoMap.get(key);
```

Comment:
can we do something like
````
table.schema().findType(key)
````

Reply:
This is a Iceberg API and we are in TruncateTableAnalyzer, hence we cant use it.

Reply:
don't we have access to the storage handler?

Reply:
We need the category of every column of the table and try to match it with the partition spec. How do you propose doing that with storage handler? Create another API?

Deserializer has good information regarding the category of the columns, which is why I have used it in this case.

Reply:
Created a `storageHandler.getColumnType()` in HiveIcebergStorageHandler. Done.